virttest/qemu_vm.py

```diff
@@ -1632,83 +1534,47 @@ def sort_key(dev):
                 devices.insert(pvpanic_dev)
 
         # Add serial console redirection
-        for serial in params.objects("serials"):
+        serials = params.objects('serials')
+        if serials:
+            self.serial_session_device = serials[0]
+            host = params.get('chardev_host', '127.0.0.1')
+            free_ports = utils_misc.find_free_ports(
+                5000, 6000, len(serials), host)
+            reg_count = 0
+        self.virtio_ports = []
```

Comment:
let's move this define in  function  '__init__'

Reply:
@xutian It actually has one in `__init__`, I removed this redundant one, thanks.

Reply:
@xutian  oh~   `__inint__`  already have, we can remove this one.